lib/src/themes/expression/parser/expression_parser.dart

```diff
@@ -96,6 +99,25 @@ class ExpressionParser {
       }
     }
     if (json is Map) {
+      // legacy categorical expressions
+      // https://docs.mapbox.com/style-spec/reference/other
+      if (json['type'] == 'categorical') {
+        final property = json['property'];
+        if (property is! String?) {
+          logger.warn(() =>
+              'Malformed categorical expression, property field must be an optional String: $json');
```

Comment:
Rather than litter the code with logs/warnings, we rely on https://github.com/greensopinion/dart-vector-tile-renderer/blob/main/lib/src/themes/expression/parser/expression_parser.dart#L124 as a catch-all.

Reply:
fixed